std/conv.d

```diff
@@ -2849,7 +2872,79 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
         enforce(ndigits, new ConvException("Error converting input"~
                         " to floating point"));
 
-        static if (real.mant_dig == 64)
+        import std.math : floatTraits, RealFormat;
+
+        static if (floatTraits!real.realFormat == RealFormat.ieeeQuadruple)
+        {
+            if (msdec)
+            {
+                /*
+                 * For quad precision, we currently allow max mantissa precision
+                 * of 64 bits, simply so we don't have to change the mantissa parser
+                 * in the code above. Feel free to adapt the parser to support full
+                 * 113 bit precision.
+                 */
+
+                // Exponent bias + 112:
+                // After shifting 112 times left, exp must be 1
+                int e2 = 0x3FFF + 112;
+
+                /*
+                 * left justify mantissa: The implicit bit (bit 112) must be 1
+                 * after this, (it is implicit and always defined as 1, so making
+                 * sure we end up with 1 at 112 means we adjust mantissa and eponent
+                 * to fit the ieee format)
+                 * For quadruple, this is especially fun as we have to use 2 longs
+                 * to store the mantissa and care about endianess...
+                 * quad_mant[0]               | quad_mant[1]
+                 * S.EEEEEEEEEEEEEEE.MMMMM .. | MMMMM .. 00000
+                 *                48       0  |
+                 */
+                ulong[2] quad_mant;
+                quad_mant[1] = msdec;
+                while ((quad_mant[0] & 0x0001_0000_0000_0000) == 0)
+                {
+                    // Shift high part one bit left
+                    quad_mant[0] <<= 1;
+                    // Transfer MSB from quad_mant[1] as new LSB
+                    quad_mant[0] |= (quad_mant[1] & 0x8000_0000_0000_0000) ? 0b1 : 0b0;
+                    // Now shift low part one bit left
+                    quad_mant[1] <<= 1;
+                    // Finally, decrease the exponent, as we increased the value
+                    // by shifting of the mantissa
+                    e2--;
+                }
+
+                ()@trusted {
+                    ulong* msw, lsw;
+                    version (LittleEndian)
+                    {
+                        lsw = &(cast(ulong*)&ldval)[0];
+                        msw = &(cast(ulong*)&ldval)[1];
+                    }
+                    else
+                    {
+                        msw = &(cast(ulong*)&ldval)[0];
+                        lsw = &(cast(ulong*)&ldval)[1];
+                    }
+
+                    // Stuff mantissa directly into double
+                    // (first including implicit bit)
+                    *msw = quad_mant[0];
+                    *lsw = quad_mant[1];
```

Comment:
Would using a union here be better? I'm a bit weary of the optimizer here, incase it just sees this as dead code (msw/lsw are set but not read).

Would probably be more clearer as well that you are changing the value of `ldval` directly.

Reply:
Probably not a problem however. I just have a slight distaste for using pointers for bit manipulation in std.math that carries over here.

Reply:
The IEEE extended and double branches used pointers as well and I'm not sure if this is relevant for CTFE-ability? I think the optimizer shouldn't be much of a problem, as the pointers are used to write to the value?

Reply:
I've seen it go wrong before when I implemented `floorImpl`. From what I recall, it couldn't trace back to the original type, and so thought that the program was modifying bits outside the range of the type it was pointing to.